src/components/utils/LayerManager.ts

```diff
@@ -18,23 +22,39 @@ export type ContentElement =
 
 export interface LayerConfig extends LayerExtendableProps {
     contentRefs?: Array<React.RefObject<ContentElement> | undefined>;
+    focusTrapContainersRefs?: React.RefObject<HTMLElement>[];
+}
+
+export interface AddResult {
+    focusTrap?: FocusTrap;
+}
+
+function isNonNullable<T>(value: T): value is NonNullable<T> {
+    return value !== null && value !== undefined;
 }
 
 class LayerManager {
     private stack: LayerConfig[] = [];
     private mouseDownTarget: HTMLElement | null = null;
+    private focusTraps = new WeakMap<LayerConfig, FocusTrap>();
```

Comment:
Do we need to pause/unpause focus trap ourself? From [documentation](https://github.com/focus-trap/focus-trap#one-at-a-time):
> Focus trap manages a queue of traps: if A activates; then B activates, pausing A; then C activates, pausing B; when C then deactivates, B is unpaused; and when B then deactivates, A is unpaused.